exercises/concept/lasagna/src/test/java/utils/ReflectionProxy.java

```diff
@@ -0,0 +1,477 @@
+package utils;
+
+import java.lang.reflect.*;
+
+import static java.lang.Class.forName;
+
+public abstract class ReflectionProxy {
```

Comment:
This class is long because features were added from time to time, in a collaborative but unorganised way. 
I made my best to document all the methods before bringing it to this repository, but I'm sure you will find a lot of things that could be improved.